src/main/java/io/aiven/kafka/connect/http/sender/AccessTokenHttpSender.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2021 Aiven Oy and http-connector-for-apache-kafka project contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.http.sender;
+
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Base64;
+import java.util.StringJoiner;
+
+import org.apache.kafka.connect.errors.ConnectException;
+
+import io.aiven.kafka.connect.http.config.HttpSinkConfig;
+
+import static io.aiven.kafka.connect.http.config.OAuth2AuthorizationMode.HEADER;
+import static io.aiven.kafka.connect.http.config.OAuth2AuthorizationMode.URL;
+
+class AccessTokenHttpSender extends AbstractHttpSender implements HttpSender {
+
+    AccessTokenHttpSender(final HttpSinkConfig config, final HttpClient httpClient) {
+        super(config, new AccessTokenHttpRequestBuilder(), httpClient);
+    }
+
+    HttpResponse<String> call() {
+        final var accessTokenRequestBodyBuilder = new StringJoiner("&");
+        accessTokenRequestBodyBuilder.add(encodeNameAndValue("grant_type", "client_credentials"));
+        if (config.oauth2ClientScope() != null) {
+            accessTokenRequestBodyBuilder.add(encodeNameAndValue("scope", config.oauth2ClientScope()));
+        }
+        setClientIdAndSecret(accessTokenRequestBodyBuilder, config);
+        return super.send(accessTokenRequestBodyBuilder.toString());
```

Comment:
the `AccessTokenHttpSender` responsibility is to call the OAuth2 URI and retrieve a token, and that should be it. 
In my opinion, the retry strategy should always be handled by the Oauth2HttpSender implementation which needs an access token. We need to separate responsibilities here otherwise we don't know when we are suppose to fail/retry and stop the thread.

Reply:
Moreover, the `send` from super class does call the `sendWithRetry` by itself.